src/anyp/Uri.cc

```diff
@@ -413,6 +423,20 @@ AnyP::Uri::parse(const HttpRequestMethod& method, const SBuf &rawUrl)
                 ++t;
                 foundPort = atoi(t);
             }
+
+            if (validateAsIpv6) {
+                // TODO: Add this check to parseHost().
```

Comment:
Please implement that TODO in this PR.

Reply:
OK, I will.

Reply:
Done in commit 95fc071. That commit also adds two new post-fromHost() assertions. I do not think adding this check to parseHost() changed parseHost() behavior, but you should check my logic -- if I am wrong, then we should adjust PR description (at least) accordingly. I am quoting that commit message below for your convenience.

----

Ban brackets in CONNECT targets sooner when !Ip::EnableIpv6

This change spills branch changes to CONNECT handling code path but
avoids making GET and CONNECT checks more different.

AFAICT, the added parseHost() check does not change the end result of
parseHost() because our fromHost() check already rejects IPv6 addresses
when IPv6 support is disabled. However, it is probably a good idea to
reject all bracketed addresses early when IPv6 support is disabled. This
clarifies the intent and protects us in case something goes wrong with
fromHost() implementation again. In the spirit of this logic, I added an
assertion to double check that fromHost() is returning an IPv6 address
when our parseHost() check expects one.